code/modules/spells/spell_types/pointed/swap.dm

```diff
@@ -0,0 +1,82 @@
+/datum/action/cooldown/spell/pointed/swap
+	name = "Swap"
+	desc = "This spell allows you to swap locations with any living being."
+	button_icon_state = "swap"
+	ranged_mousepointer = 'icons/effects/mouse_pointers/swap_target.dmi'
+
+	school = SCHOOL_TRANSLOCATION
+	cooldown_time = 30 SECONDS
+	cooldown_reduction_per_rank = 6 SECONDS
+	cast_range = 9
+	invocation = "FRO' BRT'TRO, DA!"
+	invocation_type = INVOCATION_SHOUT
+	spell_requirements = SPELL_REQUIRES_NO_ANTIMAGIC|SPELL_REQUIRES_OFF_CENTCOM
+	active_msg = "You prepare to swap locations with a target..."
+
+	smoke_type = /datum/effect_system/fluid_spread/smoke
+	smoke_amt = 0
+	/// A variable for holding the second selected target with right click.
+	var/mob/living/second_target
```

Comment:
would this be better to use weakref instead?

Reply:
I cannot play sounds to a weakref so I'll use this container instead